src/services/stake_and_mint/ProgressService.ts

```diff
@@ -59,6 +59,7 @@ export default class ProgressService {
     Logger.debug('Progress service invoked');
 
     const progressPromises = messages
+      .filter(message => message.type === MessageType.Stake)
```

Comment:
both filters can be merged to save the extra loop.

Reply:
Done!